### PR TITLE
AI - Fixed jumping and escaping (second)

### PR DESCRIPTION
**Pull request by SafwatHalaby**

### CODE REVIEW

**Comment by SafwatHalaby** [comment]
Merge?

File: src/Mobs/Monster.cpp [hunk 1]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (bearbin) [inline src/Mobs/Monster.cpp]: This space should be kept.

Comment (SafwatHalaby) [inline src/Mobs/Monster.cpp]: Code convention stuff or MS compiler stuff?

Comment (bearbin) [inline src/Mobs/Monster.cpp]: Code convention, for some reason global code search in VS needed a blank line at the start of the file. Ask @madmaxoft for more info.

Comment (SafwatHalaby) [inline src/Mobs/Monster.cpp]: Done, I'll squash the commits once I receive no further remarks for a while.

**Comment by SafwatHalaby** [comment]
The GCC build failed with a strange error which doesn't to be related to the PR. Can anyone enlighten me?

```
 if [ "$TRAVIS_MCSERVER_BUILD_TYPE" == "COVERAGE" ]; then sudo pip install cpp_coveralls; fi

The directory '/home/travis/.cache/pip/log' or its parent directory is not owned by the current user and the debug log has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want the -H flag.

The directory '/home/travis/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want the -H flag.

You are using pip version 6.0.7, however version 6.1.1 is available.

You should consider upgrading via the 'pip install --upgrade pip' command.

The directory '/home/travis/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want the -H flag.

Collecting cpp-coveralls

Downloading cpp-coveralls-0.3.6.tar.gz

Collecting requests (from cpp-coveralls)

Downloading requests-2.7.0-py2.py3-none-any.whl (470kB)

100% |################################| 471kB 720kB/s

Installing collected packages: requests, cpp-coveralls

Running setup.py install for cpp-coveralls

Installing cpp-coveralls script to /usr/local/bin

Installing coveralls script to /usr/local/bin

Successfully installed cpp-coveralls-0.3.6 requests-2.7.0
before_install.2

4.95s$ sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test

Executing: gpg --ignore-time-conflict --no-options --no-default-keyring --secret-keyring /tmp/tmp.3RMkyftd1B --trustdb-name /etc/apt/trustdb.gpg --keyring /etc/apt/trusted.gpg --primary-keyring /etc/apt/trusted.gpg --keyring /etc/apt/trusted.gpg.d//apt.postgresql.org.gpg --keyserver hkp://keyserver.ubuntu.com:80/ --recv 60C317803A41BA51845E371A1E9377A2BA9EF27F

gpg: requesting key BA9EF27F from hkp server keyserver.ubuntu.com

gpg: key BA9EF27F: public key "Launchpad Toolchain builds" imported

gpg: Total number processed: 1

gpg: imported: 1 (RSA: 1)
before_install.3

81.82s$ sudo apt-get update -qq

11.20s$ sudo apt-get install -qq g++-4.8

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-5/gcc-5-base_5.1.0-0ubuntu11~12.04.2_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-5/libgcc1_5.1.0-0ubuntu11~12.04.2_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-5/libstdc++6_5.1.0-0ubuntu11~12.04.2_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-4.8/gcc-4.8-base_4.8.1-2ubuntu1~12.04_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-4.8/libasan0_4.8.1-2ubuntu1~12.04_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-5/libatomic1_5.1.0-0ubuntu11~12.04.2_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/i/isl/libisl10_0.12.2-2~12.04_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/c/cloog/libcloog-isl4_0.18.2-1~12.04_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-5/libgomp1_5.1.0-0ubuntu11~12.04.2_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-5/libitm1_5.1.0-0ubuntu11~12.04.2_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-5/libquadmath0_5.1.0-0ubuntu11~12.04.2_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-5/libtsan0_5.1.0-0ubuntu11~12.04.2_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-4.8/cpp-4.8_4.8.1-2ubuntu1~12.04_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-4.8/libgcc-4.8-dev_4.8.1-2ubuntu1~12.04_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-4.8/gcc-4.8_4.8.1-2ubuntu1~12.04_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-4.8/libstdc++-4.8-dev_4.8.1-2ubuntu1~12.04_amd64.deb Unable to connect to ppa.launchpad.net:http:

Failed to fetch http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu/pool/main/g/gcc-4.8/g++-4.8_4.8.1-2ubuntu1~12.04_amd64.deb Unable to connect to ppa.launchpad.net:http:

E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?

The command "sudo apt-get install -qq g++-4.8" failed and exited with 100 during .

Your build has been stopped.
```

**Comment by bearbin** [comment]
I have no idea what the issue is.

**Comment by SafwatHalaby** [comment]
Pushed pointless commits to see if that helps.

**Comment by SafwatHalaby** [comment]
Looks like it helped. Squashing.

**Comment by madmaxoft** [comment]
Actually it's just easier to restart the build on the Travis page. Though I'm not sure if one has to be a member of the mc-server organization in order to be allowed to do that.

**Comment by worktycho** [comment]
I think you do have to be a member.

**Comment by SafwatHalaby** [comment]
I briefly looked for such a thing and didn't find one.

**Comment by SafwatHalaby** [comment]
I'm confused, the diff is showing 1 change only, did I accidentally base a different branch that was already merged on this one?

**Comment by SafwatHalaby** [comment]
Yes, I already smuggled these changes in the "burn" PR, oops.
Feel free to merge this then, it only adds spaces.

**Comment by bearbin** [comment]
It adds one space and removes another?

**Comment by SafwatHalaby** [comment]
Oh, sorry. Can I just close I'll just close this one and submit another for fixing spaces. I tried reverting things and I only made it all worse and I don't want to bother with fixing it all.

**Pull request closed without merging**